modules/generator/config.go

```diff
@@ -19,10 +19,11 @@ const (
 
 // Config for a generator.
 type Config struct {
-	Ring      RingConfig      `yaml:"ring"`
-	Processor ProcessorConfig `yaml:"processor"`
-	Registry  registry.Config `yaml:"registry"`
-	Storage   storage.Config  `yaml:"storage"`
+	Ring       RingConfig      `yaml:"ring"`
+	Processor  ProcessorConfig `yaml:"processor"`
+	Registry   registry.Config `yaml:"registry"`
+	Storage    storage.Config  `yaml:"storage"`
+	MaxSpanAge int64           `yaml:"max_span_age_sec"`
```

Comment:
We can make this typed
```suggestion
	MaxSpanAge time.Duration   `yaml:"max_span_age_sec"`
```
This allows users to specifcy e.g. `30s` or `5m`. It will also be easier to work with in `updatePushMetrics`.

Reply:
+1 to using a duration

regarding the name. perhaps we could choose something similar to:

https://github.com/grafana/tempo/blob/5c885f37f1a01b07db1be1f6d6ca04ce1a58113c/tempodb/wal/wal.go#L51